crates/chia-consensus/src/run_block_generator.rs

```diff
@@ -244,3 +249,62 @@ where
     ret.cost = max_cost - cost_left;
     Ok(ret)
 }
+
+pub fn get_coinspends_for_trusted_block(
+    constants: &ConsensusConstants,
+    generator: &Program,
+    refs: Vec<&[u8]>,
+    flags: u32,
+) -> Result<Vec<CoinSpend>, ValidationErr> {
+    let mut a = make_allocator(LIMIT_HEAP);
+    let mut output = Vec::<CoinSpend>::new();
+
+    let program = node_from_bytes_backrefs(&mut a, generator)?;
+    let args = setup_generator_args(&mut a, refs)?;
+    let dialect = ChiaDialect::new(flags);
+
+    let Reduction(_clvm_cost, res) = run_program(
+        &mut a,
+        &dialect,
+        program,
+        args,
+        constants.max_block_cost_clvm,
+    )?;
+
+    let (first, _rest) = a
+        .next(res)
+        .ok_or(ValidationErr(res, ErrorCode::GeneratorRuntimeError))?;
+    let mut cache = TreeCache::default();
+    let mut iter = first;
+    while let Some((spend, rest)) = a.next(iter) {
+        iter = rest;
+        let [_, puzzle, _] = extract_n::<3>(&a, spend, ErrorCode::InvalidCondition)?;
```

Comment:
we should probably be consistent with how we treat this error with line 287. If this fails, we'll abort. Which isn't necessarily unreasonable, it means the precondition was violated, because the block was invalid. But right now line 299 will silently skip the spend if it's invalid, so whatever we do, it would make sense to be consistent.